cypress/integration/pages/mediaAssetPage/helpers.js

```diff
@@ -14,3 +16,29 @@ export const getBlockData = (blockType, body) => {
 
   return getBlockByType(blocks, blockType);
 };
+
+export const hasMedia = body => {
+  const mediaTypes = ['video', 'version', 'media', 'legacyMedia'];
+  return mediaTypes.some(type => body.metadata.blockTypes.includes(type));
+};
+
+const getServiceId = body => {
+  const mediaBlock = body.promo.media;
+  const versionId = mediaBlock.versions && mediaBlock.versions[0].versionId;
+  const id = mediaBlock.externalId || mediaBlock.id;
+
+  return versionId || id;
+};
+
+export const getEmbedUrl = (body, language) => {
+  const isLegacyMedia = body.promo.media.type === 'legacyMedia';
+  const prefix = isLegacyMedia ? 'legacy' : 'cps';
+
+  return [
+    envConfig.avEmbedBaseUrl,
+    'ws/av-embeds',
+    `${prefix}${body.metadata.locators.assetUri}`,
+    getServiceId(body),
```

Comment:
This part of the URL shouldn't be the service - it should be:
- the vpid for modern maps
- the block id for legacy maps

Reply:
actually the function does look like it's returning the correct thing - it's just misnamed :)